compiler/ilgen/IlBuilder.cpp

```diff
@@ -1336,56 +1336,90 @@ IlBuilder::GreaterThan(TR::IlValue *left, TR::IlValue *right)
    return returnValue;
    }
 
-TR::IlValue *
-IlBuilder::Call(const char *functionName, int32_t numArgs, ...)
+TR::IlValue** 
+IlBuilder::processCallArgs(TR::Compilation *comp, int numArgs, va_list args)
    {
-   TR::IlValue ** argValues = (TR::IlValue **) _comp->trMemory()->allocateHeapMemory(numArgs * sizeof(TR::IlValue *));
-   va_list args;
-
-   va_start(args, numArgs);
+   TR::IlValue ** argValues = (TR::IlValue **) comp->trMemory()->allocateHeapMemory(numArgs * sizeof(TR::IlValue *));
    for (int32_t a=0;a < numArgs;a++)
       {
       argValues[a] = va_arg(args, TR::IlValue*);
       }
-   va_end(args);
-
-   return Call(functionName, numArgs, argValues);
+   return argValues;
    }
 
+/*
+ * \param computedVirtualCallAddress
+ *    It's normally an address loaded from the virtual function table
+ * \param numArgs 
+ *    Number of real arguments from the method signature
+ * \param ... 
+ *    The real arguments from the method signature
+ */
 TR::IlValue *
-IlBuilder::Call(const char *functionName, int32_t numArgs, TR::IlValue ** argValues)
+IlBuilder::ComputedVirtualCall(const char *functionName, TR::IlValue *computedVirtualCallAddress, int32_t numArgs, ...)
```

Comment:
I think an easier notation would be:
   ComputedCall(const char *functionName, int32_t numArgs, ...)

then pull off the first arg as computedCallAddress. That makes the calling sequence a bit more natural too:
b->ComputedCall("foo", 2,
b->   Load("the target entry point"),
b->   Load("parameter1"));

as opposed to the mix:
b->ComputedCall("foo",
b->   Load("the target entry point"), 1,
b->   Load("parameter 1"));

I think I prefer the first one without the numArgs value (1) "lost" in the middle.

I'm also not sure there's anything strictly "virtual" about this call, since the "virtualness" is all done by the client of the interface. That's why I removed "Virtual" from the name in the above examples.